tests/unittests/Foundation/NSSetTests.mm

```diff
@@ -148,4 +148,98 @@
     NSMutableSet* set = [NSMutableSet set];
     EXPECT_ANY_THROW([set addObject:nil]);
     EXPECT_EQ(0, [set count]);
-}
\ No newline at end of file
+}
+
+// Cannot be certain of order of elements so cannot compare directly with reference platform
+// The reference platform also has some unexpected behaviours when nesting collections, so we have opted for consistency
+OSX_DISABLED_TEST(NSSet, Description) {
+    NSSet* emptySet = [NSSet set];
+    EXPECT_OBJCEQ(@"{(\n)}", emptySet.description);
+    NSSet* nestedSet = [NSSet setWithObjects:[NSSet setWithObjects:@1, @[ @2, @"3", [NSSet setWithObject:@4] ], @5, nil], @"6", nil];
+    NSString* expected = @"{(\n"
+                          "        {(\n"
+                          "                (\n"
+                          "            2,\n"
+                          "            3,\n"
+                          "                        {(\n"
+                          "                4\n"
+                          "            )}\n"
+                          "        ),\n"
+                          "        1,\n"
+                          "        5\n"
+                          "    )},\n"
+                          "    6\n"
+                          ")}";
+    EXPECT_OBJCEQ(expected, nestedSet.description);
+}
+
+TEST(NSSet, MakeObjectsPerformSelector) {
+    NSSet* set = [NSSet setWithObjects:@[ @1, @2, @3 ], [NSSet setWithObjects:@4, @5, nil], @[ @6 ], nil];
+    __block NSMutableSet* otherSet = [NSMutableSet set];
+    [set makeObjectsPerformSelector:@selector(enumerateObjectsUsingBlock:)
+                         withObject:^(id obj, BOOL* stop) {
+                             [otherSet addObject:obj];
```

Comment:
No this is making the elements of the set, which are sets and arrays, perform `enumerateObjectsUsingBlock` with the block as the `withObject` object.